api.go

```diff
@@ -59,8 +61,39 @@ func NewAPI() *API {
 	}
 }
 
+// functionStates specifies the api functions that are valid for each
+// cluster state.
+var functionStates = map[string]map[apiFunc]struct{}{
+	ClusterStateStarting: functionCommon,
+	ClusterStateNormal:   appendMap(functionCommon, functionNormal),
+	ClusterStateResizing: appendMap(functionCommon, functionResizing),
+}
+
+func appendMap(a, b map[apiFunc]struct{}) map[apiFunc]struct{} {
+	r := make(map[apiFunc]struct{})
+	for k, v := range a {
+		r[k] = v
+	}
+	for k, v := range b {
+		r[k] = v
+	}
+	return r
+}
+
+func (api *API) validate(f apiFunc) error {
+	state := api.Cluster.State()
+	if _, ok := functionStates[state][f]; ok {
```

Comment:
I think I'd rename `functionStates` to `validAPIMethods` or `validAPIFuncs` if we're going to stick with function terminology everywhere (e.g. the `apiFunc` type)

Reply:
oops. yeah, I forgot to change `func` to `method`. thanks for the reminder. I'll rename this while I'm at it.